pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -191,13 +192,18 @@ func (p *LaunchTemplateProvider) getSecurityGroupIds(ctx context.Context, provis
 	return securityGroupIds, nil
 }
 
-func (p *LaunchTemplateProvider) getUserData(ctx context.Context, provisioner *v1alpha3.Provisioner, constraints *Constraints) (string, error) {
+func (p *LaunchTemplateProvider) getUserData(ctx context.Context, provisioner *v1alpha3.Provisioner, constraints *Constraints, instanceTypes []cloudprovider.InstanceType) (string, error) {
+	var containerRuntimeArg string
+	if !NeedsDocker(instanceTypes) {
```

Comment:
How hard is it to remove the docker dependency?

Reply:
Sorry, I do not understand the question, can you clarify?

Reply:
How hard is it to use containerd for GPUs? Happy if the answer is "out of scope for now"

Reply:
My understanding is, it's not hard. The EKS optimized AMI does already support it for Nvidia GPUs, just not for Neurons.